src/Common/Config/CMakeLists.txt

```diff
@@ -1,8 +1,18 @@
-include(${ClickHouse_SOURCE_DIR}/cmake/dbms_glob_sources.cmake)
+set (SRCS
+    AbstractConfigurationComparison.cpp
+    ConfigProcessor.cpp
+    configReadClient.cpp
+    ConfigReloader.cpp
+)
 
-add_headers_and_sources(clickhouse_common_config .)
+add_library(clickhouse_common_config ${SRCS})
 
-add_library(clickhouse_common_config ${clickhouse_common_config_headers} ${clickhouse_common_config_sources})
-
-target_link_libraries(clickhouse_common_config PUBLIC common PRIVATE clickhouse_common_zookeeper string_utils PUBLIC ${Poco_XML_LIBRARY} ${Poco_Util_LIBRARY})
 target_include_directories(clickhouse_common_config PUBLIC ${DBMS_INCLUDE_DIR})
+target_link_libraries(clickhouse_common_config
+    PUBLIC
+        common
+    PRIVATE
+        clickhouse_common_zookeeper
+        Poco::XML
```

Comment:
`Poco::XML` should be `PUBLIC`, otherwise build of `utils/config-processor` will fail (and some others)